projects/vdk-plugins/vdk-singer/src/vdk/plugin/singer/singer_command.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import click
+import requests
+from vdk.internal.control.utils import cli_utils
+from vdk.internal.control.utils import output_printer
+
+
+def is_likely_singer_tap(package_name: str) -> bool:
+    response = requests.get(f"https://pypi.org/pypi/{package_name}/json")
+    if response.status_code == 200:
+        return "singer" in (response.text.lower())
+
+
+def find_packages_starting_with(prefix: str) -> list[str]:
+    url = "https://pypi.org/simple/"
```

Comment:
Is this getting a list of all packages for packages which start with tap? 
seems like it will need a lot of ram to run?

Reply:
Ram is about 150-200 MBs according to psutil.memory_info. As a standalone /adhoc/ command is fine. 

The pypi.org/simple is just a list of package names so it's not that huge.